packages/jsapi-utils/src/ConnectionUtils.ts

```diff
@@ -34,10 +34,12 @@ export function fetchVariableDefinition(
      */
     function handleFieldUpdates(changes: VariableChanges): void {
       const definition = changes.created.find(def => def.title === name);
+      clearTimeout(timeoutId);
+      removeListener?.();
       if (definition != null) {
-        clearTimeout(timeoutId);
-        removeListener?.();
         resolve(definition);
+      } else {
+        reject(new TimeoutError(`Variable ${name} not found`));
```

Comment:
So in this case, we do _not_ want a `TimeoutError`, since this isn't being triggered from a `Timeout`. Just reject with `new Error(...)` instead:
```suggestion
        reject(new Error(`Variable ${name} not found`));
```

Reply:
We could modify the TimeoutError above to be different as well, e.g. `TimeoutError(\`Timeout looking for variable ${name}\`)`